docs/source/resources/graphql-glossary.md

```diff
@@ -0,0 +1,356 @@
+      if (loading) return null;
+      if (error) return 'Error!';
+      return (
+        <img src={data.dog.displayImage} />
+      );
+    }}
+  </Query>
+);`;
+```
+
+<h2 id="query-whitelisting">Query whitelisting</h2>
+<p>A technique for preventing unwanted attacks by maintaining a list of approved queries that are allowed in your application. Any query not present in the list that is run against the server will not be allowed. [Automatic Persisted Queries](../guides/performance.html#automatic-persisted-queries) is a feature of Apollo Server 2 that enables query whitelisting and persisted queries.</p>
+
+<h2 id="resolver">Resolver</h2>
+<p>A function that connects schema fields and types to various backends. Resolvers provide the instructions for turning a GraphQL operation into data. It can retrieve or write data from either an SQL, a No-SQL, graph database, a micro-service or a REST API. Resolvers can also return strings, ints, null, and other primitives.</p>
+
+```js
+...
+const resolvers = {
+  Query: {
+    author(root, args, context, info) {
+      return find(authors, { id: args.id });
+    },
+  },
+  Author: {
+    books(author) {
+      return filter(books, { author: author.name });
+    },
+  },
+};
+```
+
+
+<h2 id="schema">Schema</h2>
+<p>A GraphQL [schema](https://www.apollographql.com/docs/apollo-server/essentials/schema.html) is at the center of any GraphQL server implementation and describes the functionality available to the clients which connect to it.</p>
+
+
+<h2 id="schema-definition-language">Schema Definition Language (SDL)</h2>
+<p>The syntax for writing GraphQL Schemas. It is otherwise known as Interface Definition Language. It is the lingua franca shared by all for building GraphQL APIs regardless of the programming language chosen.</p>
+
+```js
+type Author {
+  id: Int!
+  firstName: String
+  lastName: String
+  posts: [Post]
+}
+type Post {
+  id: Int!
+  title: String
+  author: Author
+  votes: Int
+}
+type Query {
+  posts: [Post]
+  author(id: Int!): Author
+}
+```
+
+<h2 id="schema-first-development">Schema first development</h2>
+<p>A [development approach](https://www.apollographql.com/docs/fundamentals/tips.html#schema) for designing and building modern UIs that involves the frontend and backend teams agreeing on a Schema first, which serves as a contract between the UI and the backend before any API engineering happens.</p>
+
+
+<h2 id="schema-registry">Schema registry</h2>
+<p>A central source of truth for your schema in Apollo Engine. It enables schema registration, schema validation, tracking of detailed schema changes e.g. types added, fields added, fields deprecated and looking up previous versions of schema.</p>
+
+
+<h2 id="schema-versioning">Schema versioning</h2>
+<p>Refers to the ability to have different versions of your Schema. This allows for reversible changes to be made to the Schema. The Apollo CLI is a tool that provides and manages Schema versioning with Engine.</p>
```

Comment:
"Refers to the need to evolve a schema over time. As a schema evolves, there is a potential for introducing breaking changes to clients. The Apollo CLI assists schema evolution by validating schema changes and checking for breaking changes using Apollo Engine. Read more in the [versioning guide](../guides/versioning.html)"

Reply:
This is perfect!. Thanks @JakeDawkins